app/src/main/java/ru/kryu/playlistmaker/settings/ui/activity/SettingsActivity.kt

```diff
@@ -0,0 +1,60 @@
+package ru.kryu.playlistmaker.settings.ui.activity
+
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.ViewModelProvider
+import ru.kryu.playlistmaker.R
+import ru.kryu.playlistmaker.app.App
+import ru.kryu.playlistmaker.databinding.ActivitySettingsBinding
+import ru.kryu.playlistmaker.settings.ui.view_model.DarkThemeState
+import ru.kryu.playlistmaker.settings.ui.view_model.SettingsViewModel
+
+class SettingsActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivitySettingsBinding
+    private lateinit var viewModel: SettingsViewModel
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivitySettingsBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        viewModel = ViewModelProvider(
+            this,
+            SettingsViewModel.getViewModelFactory()
+        )[SettingsViewModel::class.java]
+        viewModel.darkThemeStateLiveData.observe(this) { render(it) }
+
+        binding.buttonBackSettings.setOnClickListener {
+            finish()
+        }
+
+        binding.themeSwitcher.isChecked =
+            (viewModel.darkThemeStateLiveData.value == DarkThemeState.STATE_DARK)
+        binding.themeSwitcher.setOnCheckedChangeListener { swither, checked ->
+            viewModel.changeState(checked)
+        }
+
+        binding.shareFrame.setOnClickListener {
+            viewModel.doActionSend(getString(R.string.url_practicum))
```

Comment:
🍏 Рекомендую не давать вьюмодели конкретных указаний от активити. Достаточно сообщить, что нажата определенная кнопка, а вьюмодель уже сама должна решить, что ей делать.
```suggestion
            viewModel.onShareClick()
```